06.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  STRIKE_MARK = 'X'
+
+  attr_reader :frames
+
+  def initialize(shot_marks)
+    delimited_marks = parse_marks(shot_marks.split(','))
+    @frames = delimited_marks.map { |m| Frame.new(m) }
+  end
+
+  def score
+    pin_score = @frames.map(&:score).sum
```

Comment:
```suggestion
    @frames.sum { |frame| frame.calc_score(@frames) }
```

こんな感じでframe自身に「計算よろしく！」って頼んだら各frameが自分の得点を計算してくれるようにするとオブジェクト指向らしくていいかも

Reply:
frameに@framesを渡すということで、frameの実装を全体的に見直すイメージでしょうか？
frameは、各フレームに閉じた状態をもつ（frame間の橋渡しは責務外）と思っておりまして、修正方法がわからず...

Reply:
「frameは、各フレームに閉じた状態をもつ」という考え方はOKです。

ただし、ここでは別にframeが他のframeに関する責務を持っているわけではなく、単に引数として渡してもらうことを想定しているので、責務ではないと考えて良いと思います。
（あるframeが他のframeをnewしたり、他のframeをスコアを変更したりすることがあるなら話は別ですが）